instrumentation/opentelemetry-instrumentation-celery/src/opentelemetry/instrumentation/celery/__init__.py

```diff
@@ -118,7 +119,8 @@ def _trace_prerun(self, *args, **kwargs):
             return
 
         request = task.request
-        tracectx = propagators.extract(carrier_extractor, request) or None
+        getter = HelperGetter(carrier_extractor, DictGetter.keys)
```

Comment:
1. getter should be initialized once per module as they are stateless and don't need to be re-created for every span.
2. if something ever called `getter.keys()` on this instance, it would panic because `task.request` is most-likely not a dictionary according to the implementation of `carrier_extractor`. It looks like it is an object and carrier_extractor is using getattr to extract carrier values from object attributes. I'm not sure what an ideal implementation of keys would be here. It may be to have it as noop so it always returns an empty list or it may be to return every attribute the carrier has. A third option might be to return well-known carrier keys when present on the object as attributes.

Reply:
@owais I've updated the getter as subclass of DictGetter and overridden the keys method to  return an empty list